src/daft-sql/src/planner.rs

```diff
@@ -1959,6 +1964,75 @@ impl<'a> SQLPlanner<'a> {
                     .collect::<SQLPlannerResult<Vec<_>>>()?;
                 DataType::Struct(fields)
             }
+            SQLDataType::Custom(name, properties) => match name.to_string().as_str() {
```

Comment:
need to case-normalize to lowercase or some other case-insensitive match strategy